src/lib/reasoners/bitv.ml

```diff
@@ -1451,11 +1443,17 @@ module Shostak(X : ALIEN) = struct
   let solve r1 r2 pb =
     Sig.{pb with sbt = List.rev_append (solve_bis r1 r2) pb.sbt}
 
-  (* Pop the first bit, raises [Not_found] if there is no first bit *)
+  (* Pop the first bit, raises [Not_found] if there is no first bit.
+
+     Note that the returned bv has an incorrect size. *)
```

Comment:
How is the size incorrect?

Reply:
Because the underlying value is not resized, the size is just changed (for instance we could have `0b1111` with `size = 1` which is "wrong" in other parts of the code). So the high bits must be ignored, which `pop_bit` does.